backend/danswer/indexing/chunker.py

```diff
@@ -255,31 +262,71 @@ def chunk_document(
     return chunks
 
 
-def split_chunk_text_into_mini_chunks(
-    chunk_text: str, mini_chunk_size: int = MINI_CHUNK_SIZE
+def _split_chunk_text_into_mini_chunks(
+    chunk_text: str, embedder: IndexingEmbedder, mini_chunk_size: int = MINI_CHUNK_SIZE
 ) -> list[str]:
     """The minichunks won't all have the title prefix or metadata suffix
     It could be a significant percentage of every minichunk so better to not include it
     """
     from llama_index.text_splitter import SentenceSplitter
 
-    token_count_func = get_default_tokenizer().tokenize
+    token_count_func = get_tokenizer(
+        model_name=embedder.model_name,
+        provider_type=embedder.provider_type,
+    ).tokenize
     sentence_aware_splitter = SentenceSplitter(
         tokenizer=token_count_func, chunk_size=mini_chunk_size, chunk_overlap=0
     )
 
     return sentence_aware_splitter.split_text(chunk_text)
 
 
+def extract_chunk_texts_from_doc_aware_chunk(
```

Comment:
this is now an intermediate step between chunking and embedding

this may be able to be done at the end of the chunking function itself (and therefor kept entirely out of the indexing_pipline)